source/simulator/stokes_matrix_free.cc

```diff
@@ -249,6 +249,11 @@ namespace aspect
           ptmp.reinit(src);
         }
 
+      // This needs to be done explicitly, as GMRES does not
+      // initialize the data of the vector dst before calling
+      // us. Otherwise we might use random data as our initial guess.
+      dst = 0.0;
```

Comment:
This is of course a feature: If you start with the solution vector, GMRES should be done after one (or zero) iterations.